spec/index.bs

```diff
@@ -3520,9 +3609,83 @@ interface mixin GPUProgrammablePassEncoder {
 };
 </script>
 
+{{GPUProgrammablePassEncoder}} has the following internal slots:
```

Comment:
This is the exact same as for GPUCommandEncoder. WDYT of having a GPUEncoderBase that has this functionality as well as things like timestamp queries?

Reply:
That came up when I was talking to @kainino0x about this. There are some minor differences in validation that would need to be taken into account, but otherwise it doesn't seem like a bad idea.